src/services/utilities.ts

```diff
@@ -2335,12 +2335,12 @@ export function createTextSpanFromNode(node: Node, sourceFile?: SourceFile, endN
 }
 
 /** @internal */
-export function createTextSpanFromStringLiteralLikeContent(node: StringLiteralLike) {
+export function createTextSpanFromStringLiteralLikeContent(node: StringLiteralLike, position?: number) {
     let replacementEnd = node.getEnd() - 1;
     if (node.isUnterminated) {
         // we return no replacement range only if unterminated string is empty
         if (node.getStart() === replacementEnd) return undefined;
-        replacementEnd = node.getEnd();
+        replacementEnd = position !== undefined && position <= node.getEnd() ? position : node.getEnd();

```

Comment:
There are only two callers of `createTextSpanFromStringLiteralLikeContent`, and neither of them omit the `position` argument, so just make it required (and document what `position` is supposed to mean)

Reply:
Also, do you have any tests for when `position` is strictly greater than the end of the node? I don't think I'm seeing any.

Reply:
Ah, that was old code, thanks for catching it! (I tried to make `position` optional, but it ended up being needed in more places)

Reply:
It shouldn't be possible _in a completion_ for `position > node end` but I will add that check incase `createTextSpanFromStringLiteralLikeContent` will be used in a non-completion context in the future.

Reply:
The only thing I can think of is possibly an unterminated string literal on the previous line

```ts
let s: "string" = "s
/*$*/
```

Reply:
Completions won't be triggered on new lines -- so 
```typescript
type LongString = `asdf
string
line`

let s: LongString = `asdf
/**/
  ^ no suggestions

let r: LongString = `asdf/**/
                          ^ completes to `asdf\nstring\nline`
```